src/main/java/kitchenpos/menu/application/ProductService.java

```diff
@@ -0,0 +1,36 @@
+package kitchenpos.menu.application;
+
+import kitchenpos.menu.domain.Product;
+import kitchenpos.menu.domain.ProductRepository;
+import kitchenpos.menu.dto.ProductRequest;
+import kitchenpos.menu.dto.ProductResponse;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class ProductService {
+    private final ProductRepository productRepository;
+
+    public ProductService(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @Transactional
+    public ProductResponse create(final ProductRequest productRequest) {
+        Product saved = productRepository.save(productRequest.toEntity());
+        return ProductResponse.of(saved);
+    }
+
+    public Product findById(final Long productId){
+        return productRepository.findById(productId)
+                .orElseThrow(IllegalArgumentException::new);
+    }
+
+    public List<ProductResponse> list() {
+        List<Product> all = productRepository.findAll();
+        return all.stream().map(ProductResponse::of).collect(Collectors.toList());
+    }
```

Comment:
```suggestion
    @Transactional(readOnly = true)
    public List<ProductResponse> list() {
        List<Product> all = productRepository.findAll();
        return all.stream().map(ProductResponse::of).collect(Collectors.toList());
    }
```
`@Transactional(readOnly = true)`를 붙이는것과
트랜잭션을 안붙이는것의 차이는 무엇일까요? 🤔

Reply:
Transactional을 안 붙이면 영속성 컨텍스트가 없어서 조회가 되더라도 준영속 상태 이고요, 변경감지도 당연히 안되고 지연 로딩을 할 수가 없습니다.  Transactional(readOny = true)를 붙이면 영속 상태이고 지연 로딩을 사용할 수 있지만 변경은 할 수 없습니다